src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -466,6 +470,24 @@ bool InputProcessor::checkForUnsupportedObjects(EnergyPlusData &state)
     return errorsFound;
 }
 
+void InputProcessor::isInputObjectUsed(EnergyPlusData &state)
+{
+    // there is a need to know if certain object are used in the simulation
+    // this may not be the best place to do this but it's a start at early reading of input data
+    // this concept could grow to include reading all inputs prior to the start of the simulation so all data is available
+    // once inputs are processed, read in all getInputs. They will be read early and will not be read again so there is no duplication
+
+    // for example, AirLoopHVAC:DOAS systems autosize on weather data, and WeatherManager processes that data
+    // there is no need to process that data if there are no DOAS used in the simulation
+    state.dataInputProcessing->AirLoopHVACDOASUsedInSim =
+        state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "AirLoopHVAC:DedicatedOutdoorAirSystem") > 0;
+    EMSManager::CheckIfAnyEMS(state);
```

Comment:
So this shouldn't call back _out_ to the EMSManager.  The input processor should only rely on worker routines to develop input structures.  The CheckIfAnyEMS is also a misnamed function, because it doesn't just check if there are any inputs, it assigns a whole bunch of values.  

I need to understand the intent of collecting these, because I could imagine a few ways to make it work better.  I'll ping you separately to discuss it a bit more so we can get this wrapped up.

Reply:
The single intent of this function was to determine if AirLoopHVAC:DedicatedOutdoorAirSystem objects exist in the input. I figure we will need to know if other objects also exist so this function was created for that purpose. It seemed reasonable to also include other "check" functions in SimulationManager here since they are called in the same area of code where this function is called (whether they set a flag or get more data). I don't care if these are moved back to where they were, literaly 3 lines up from this call. I think from just seeing this little example you should have a good idea of what the intent is here, to call for information that would not be available until later in the call order (the AirLoopHVACDOASUsedInSim flag that is, not the other Check functions that were already called here).

![image](https://user-images.githubusercontent.com/5778156/181847116-4d7f9852-0b72-4b61-92cc-ba3bb3a93ea0.png)